### ISSUE TITLE
repoquery --tree

### ISSUE THREAD

**Issue by j-mracek** [rpm-software-management/dnf5#913]
[amatej] Perhaps this would be better suited as a plugin?

**Comment by m-blaha** [comment]
I agree that a plugin would be a better solution. The semi-graphical output of `repoquery --tree` does not align with other repoquery switches, all of which produce a plain list of packages.

**Comment by m-blaha** [comment]
> I agree that a plugin would be a better solution. The semi-graphical output of `repoquery --tree` does not align with other repoquery switches, all of which produce a plain list of packages.

It might even be only one `deptree` plugin covering both `--tree` and `--deplist` functionality.

**Comment by jrom99** [comment]
Is there a current suggestion to list dependencies recursively up and down the tree? I had to recently debug why `python3-pandas` was installed and couldn't figure the best approach.

**Comment by kontura** [comment]
> Is there a current suggestion to list dependencies recursively up and down the tree? I had to recently debug why `python3-pandas` was installed and couldn't figure the best approach.

There is the repoquery `--recursive` option. To go down: `dnf rq --recursive --providers-of=requires python3-pandas` and up would be: `dnf rq --recursive --whatrequires python3-pandas`

**Comment by ppisar** [comment]
> I had to recently debug why python3-pandas was installed

Using _dnf repoquery_ for this purpose is wrong because repoquery searches repositories, while what you have installed is unrelated to them. Correct approach to searching RPM database with _rpm_ tool.

**Comment by kontura** [comment]
That is a good point, possibly the `--installed` repoquery flag could be added to the queries to search only the installed packages.
